### ISSUE TITLE
Number of acked messages does not correspond to metrics in Stackdriver

### ISSUE THREAD

**Issue by Marigold** [GoogleCloudPlatform/google-cloud-python#4214]
I have this trivial script that just consumes messages from PubSub
```python
from google.cloud import pubsub

subscriber = pubsub.SubscriberClient()
subscription_path = 'projects/myproject/subscriptions/mysub'

def callback(msg):
    msg.ack()

subscriber.subscribe(subscription_path, callback=callback)
```
I'd expect that all messages would be acked without any problem, but when I look at Stackdriver, I see a huge number of `Deadline Operations` and very few `StreamingPull Acknowledge Operations`. See screenshot of Stackdriver https://imgur.com/Z4ZK2l1 (couldn't upload it here for unknown reasons).

Is this expected? Is it fault of Stackdriver or can there be a problem with acking?

Tested on Google Dataproc with `google-cloud-pubsub-0.28.4`.

Thanks!

**Comment by Marigold** [comment]
Forgot to mention that this happens for topic that has 7 days old messages, could that be an issue? I tried it on queue that is not full and the difference between acked and pulled operations were about 10% (way lower than in previous case, but still unexpected).

**Comment by lukesneeringer** [comment]
This looks like it might be related to [this question on Stack Overflow](https://www.google.com/url?q=https://stackoverflow.com/questions/46374731/pubsub-ackd-messages-continuously-re-deliver-every-10s-python-beta-client&sa=D&usg=AFQjCNFwYx0Lyowuqf7m6lMNpRm_QoWTIA).

I will look into this further.

**Comment by Marigold** [comment]
Thanks. We have a similar use case to batching https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4141 and weren't able to get even your sample code working (ratio of acked / pulled messages was extremely low). Had to switch to 0.27.0 with sync pulls which work just fine.

**Comment by dhermes** [comment]
I am going to close this, since it seems very similar to other reported issue and the `Consumer.pause()` and `Consumer.resume()`, which was resolved by #4558.

@Marigold If I'm closing this in error, I'm happy to work with you to get a case we can reproduce. (Having to wait until messages are 7 days queued will obviously take longer to reproduce :grinning:.)
